src/alchemlyb/convergence/convergence.py

```diff
@@ -137,3 +138,135 @@ def forward_backward_convergence(df_list, estimator='MBAR', num=10, **kwargs):
          'data_fraction': [i / num for i in range(1, num + 1)]})
     convergence.attrs = df_list[0].attrs
     return convergence
+
+def _cummean(vals, out_length):
+    '''The cumulative mean of an array.
+
+    This function computes the cumulative mean and shapes the result to the
+    desired length.
+
+    Parameters
+    ----------
+    vals : numpy.array
+        The one-dimensional input array.
+    out_length : int
+        The length of the output array.
+
+    Returns
+    -------
+    numpy.array
+        The one-dimensional input array with length of total.
+
+    Note
+    ----
+    If the length of the input series is smaller than the ``out_length``, the
+    length of the output array is the same as the input series.
+
+
+    .. versionadded:: 1.0.0
+
+    '''
+    in_length = len(vals)
+    if in_length < out_length:
+        out_length = in_length
+    block = in_length // out_length
+    reshape = vals[: block*out_length].reshape(block, out_length)
+    mean = np.mean(reshape, axis=0)
+    result = np.cumsum(mean) / np.arange(1, out_length+1)
+    return result
+
+def R_c(series, precision=0.01, diff=2):
+    '''Generate the convergence criteria R_c for a single simulation.
+
+    The input will be pandas.Series generated by
+    :func:`~alchemlyb.preprocessing.subsampling.decorrelate_u_nk` or
+    :func:`~alchemlyb.preprocessing.subsampling.decorrelate_dhdl`.
+
+    The output will the float R_c. R_c = 0 indicates that the system is well
+    equilibrated right from the beginning while R_c = 1 signifies that the
+    whole trajectory is not equilibrated.
+
+    Parameters
+    ----------
+    series : pandas.Series
+        The input energy array.
+    precision : float
+        The precision of the output R_c.
+    diff : float
+        Tolerance of the convergence check in kcal/mol.
+
+    Returns
+    -------
+    float
+        Convergence time fraction.
+
+    Note
+    ----
+    This function tries to compute R_c from equation 16 from
+    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8397498/#FD16. The code is
+    modified based on Shujie Fan's (@VOD555) work.
+
+
+    .. versionadded:: 1.0.0
+    '''
+    series = to_kcalmol(series)
+    array = series.to_numpy()
+    out_length = int(1 / precision)
+    g_forward = _cummean(array, out_length)
+    g_backward = _cummean(array[::-1], out_length)
+    length = len(g_forward)
+    # Final value
+    g = g_forward[-1]
+    conv = ((g_forward>g-diff) & (g_forward<g+diff)) & ((g_backward>g-diff)&(g_backward<g+diff))
+    for i in range(out_length):
+        if all(conv[i:]):
+            return i / length
```

Comment:
It's better to also return `g_forward` and `g_backward` so that one can draw the convergence plot with these two series.